python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -0,0 +1,327 @@
+
+    gpu_data = {worker:part for worker, part in worker_map}
+
+    yield wait(gpu_data)
+
+    raise gen.Return(gpu_data)
+
+
+class RandomForestRegressor:
+    """
+    Implements a multi-GPU Random Forest regressor model which fits multiple decision
+    tree classifiers in an ensemble.
+
+    Please check the single-GPU implementation of Random Forest regressor for more information
+    about the algorithm.
+
+     Parameters
+    -----------
+    n_estimators : int (default = 10)
+                   number of trees in the forest.
+    handle : cuml.Handle
+             If it is None, a new one is created just for this class.
+    split_algo : int (default = 1)
+                 0 for HIST, 1 for GLOBAL_QUANTILE and 2 for SPLIT_ALGO_END
+                 The type of algorithm to be used to create the trees.
+    split_criterion: int (default = 2)
+                     The criterion used to split nodes.
+                     0 for GINI, 1 for ENTROPY,
+                     2 for MSE, 3 for MAE and 4 for CRITERION_END.
+                     0 and 1 not valid for regression
+    bootstrap : boolean (default = True)
+                Control bootstrapping.
+                If set, each tree in the forest is built
+                on a bootstrapped sample with replacement.
+                If false, sampling without replacement is done.
+    bootstrap_features : boolean (default = False)
+                         Control bootstrapping for features.
+                         If features are drawn with or without replacement
+    rows_sample : float (default = 1.0)
+                  Ratio of dataset rows used while fitting each tree.
+    max_depth : int (default = -1)
+                Maximum tree depth. Unlimited (i.e, until leaves are pure),
+                if -1.
+    max_leaves : int (default = -1)
+                 Maximum leaf nodes per tree. Soft constraint. Unlimited,
+                 if -1.
+    max_features : int or float or string or None (default = 'auto')
+                   Ratio of number of features (columns) to consider
+                   per node split.
+                   If int then max_features/n_features.
+                   If float then max_features is a fraction.
+                   If 'auto' then max_features=n_features which is 1.0.
+                   If 'sqrt' then max_features=1/sqrt(n_features).
+                   If 'log2' then max_features=log2(n_features)/n_features.
+                   If None, then max_features=n_features which is 1.0.
+    n_bins :  int (default = 8)
+              Number of bins used by the split algorithm.
+    min_rows_per_node : int or float (default = 2)
+                        The minimum number of samples (rows) needed
+                        to split a node.
+                        If int then number of sample rows
+                        If float the min_rows_per_sample*n_rows
+    accuracy_metric : string (default = 'mse')
```

Comment:
we use a string here, making a stronger case for a string in `split_criterion` :)

Reply:
Following SG RF.